mne/stats/regression.py

```diff
@@ -134,3 +134,156 @@ def _fit_lm(data, design_matrix, names):
+    events : instance of Events
+        An n x 3 matrix, where the first column corresponds to samples in raw.
+    event_id : dict
+        As in Epochs; a dictionary where the values may be integers or
+        list-like collections of integers, corresponding to the 3rd column of
+        events, and the keys are condition names.
+    tmin : int | dict | None
+        If int, gives the lower limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types; for missing values or
+        None, the default (-.1) is used.
+    tmax : int | dict
+        If int, gives the upper limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types; for missing values or
+        None, the default (-.1) is used.
+    covariates : dict-like | None
+        If dict-like, values have to be array-like and of the same length as
+        the columns in ```events```. Keys correspond to additional event
+        types/conditions to be estimated and are matched with the time points
+        given by the first column of ```events```. If None, only binary events
+        (from event_id) are used.
+    reject : bool | dict
+        Activate rejection parameters based on peak-to-peak amplitude in
+        continuously selected subwindows. If reject is None, no rejection is
+        done. If True, the following default is employed:
+            reject = dict(grad=4000e-12, # T / m (gradiometers)
+                          mag=4e-11, # T (magnetometers)
+                          eeg=40e-5, # uV (EEG channels)
+                          eog=250e-5 # uV (EOG channels)
+                         )
+        If dict, keys are types ('grad' | 'mag' | 'eeg' | 'eog' | 'ecg') and
+        values are the maximal peak-to-peak values to select rejected epochs.
+    tstep : None | float
+        If reject is not None, length of windows for peak-to-peak detection.
+
+    Returns
+    -------
+    ev_dict : dictionary
+        A dictionary where the keys correspond to conditions and the values are
+        Evoked objects with the rE[R/F]Ps. These can be used exactly like any
+        other Evoked object, including e.g. plotting or statistics.
+    """
+    sfreq = raw.info["sfreq"]
+    data, times = raw[:]
+    conds = list(event_id.keys())
+    if covariates is not None:
+        conds += list(covariates.keys())
+
+    if isinstance(tmin, (float, int)):
+        tmin = {cond: int(tmin * sfreq) for cond in conds}
+    else:
+        tmin = {cond: int(tmin.get(cond, -.1) * sfreq) for cond in conds}
+    if isinstance(tmax, (float, int)):
+        tmax = {cond: int(tmax * sfreq) for cond in event_id.keys()}
+    else:
+        tmax = {cond: int(tmax.get(cond, 1) * sfreq) for cond in conds}
```

Comment:
@jona-sassenhagen dict comprehension syntax is not supported in python 2.6
you need to use the dict function here with tuple inputs.